--- a/drivers/crypto/inside-secure/eip93/eip93-aead.c
+++ b/drivers/crypto/inside-secure/eip93/eip93-aead.c
@@ -135,7 +135,7 @@ static int eip93_aead_setkey(struct cryp
 	struct crypto_aes_ctx aes;
 	struct sa_record *sa_record = ctx->sa_record_out;
 	u32 nonce = 0;
-	int ret;
+	int ret = -EINVAL;
 
 	if (crypto_authenc_extractkeys(&keys, key, len))
 		return -EINVAL;
@@ -152,26 +152,18 @@ static int eip93_aead_setkey(struct cryp
 	switch ((ctx->flags & EIP93_ALG_MASK)) {
 	case EIP93_ALG_DES:
 		ret = verify_aead_des_key(ctfm, keys.enckey, keys.enckeylen);
-		if (ret)
-			return ret;
-
 		break;
 	case EIP93_ALG_3DES:
 		if (keys.enckeylen != DES3_EDE_KEY_SIZE)
 			return -EINVAL;
 
 		ret = verify_aead_des3_key(ctfm, keys.enckey, keys.enckeylen);
-		if (ret)
-			return ret;
-
 		break;
 	case EIP93_ALG_AES:
 		ret = aes_expandkey(&aes, keys.enckey, keys.enckeylen);
-		if (ret)
-			return ret;
-
-		break;
 	}
+	if (ret)
+		return ret;
 
 	ctx->blksize = crypto_aead_blocksize(ctfm);
 	dma_sync_single_for_cpu(ctx->eip93->dev, ctx->sa_record_base_in,
