diff -ruN e0/drivers/crypto/inside-secure/eip93/eip93-aead.c e1/drivers/crypto/inside-secure/eip93/eip93-aead.c
--- e0/drivers/crypto/inside-secure/eip93/eip93-aead.c	2026-02-15 10:40:11.312106200 +0100
+++ e1/drivers/crypto/inside-secure/eip93/eip93-aead.c	2026-02-16 16:36:10.029450000 +0100
@@ -37,6 +37,12 @@
 	eip93_unmap_dma(eip93, rctx, req->src, req->dst);
 	eip93_handle_result(eip93, rctx, req->iv);
 
+	if (err == EIP93_HW_ERR_AUTH_FAILED)
+		err = -EBADMSG;
+	/* let software handle anti-replay errors */
+	if (err == EIP93_HW_ERR_REPLAY)
+		err = 0;
+
 	aead_request_complete(req, err);
 }
 
diff -ruN e0/drivers/crypto/inside-secure/eip93/eip93-main.h e1/drivers/crypto/inside-secure/eip93/eip93-main.h
--- e0/drivers/crypto/inside-secure/eip93/eip93-main.h	2026-02-15 10:40:32.881664700 +0100
+++ e1/drivers/crypto/inside-secure/eip93/eip93-main.h	2026-02-16 16:32:53.171983100 +0100
@@ -14,6 +14,9 @@
 #include <linux/bitfield.h>
 #include <linux/interrupt.h>
 
+#define EIP93_HW_ERR_AUTH_FAILED	1 /* Authentication/integrity check failed */
+#define EIP93_HW_ERR_REPLAY		4 /* Anti-replay check error */
+
 #define EIP93_RING_BUSY_DELAY		500
 
 #define EIP93_RING_NUM			512
