--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
@@ -633,19 +633,12 @@ static void brcmf_common_pd_remove(struc
 	if (brcmfmac_pdata->power_off)
 		brcmfmac_pdata->power_off();
 }
-#if LINUX_VERSION_IS_LESS(6,13,0)
-static int bp_brcmf_common_pd_remove(struct platform_device *pdev) {
-	brcmf_common_pd_remove(pdev);
-
-	return 0;
-}
-#endif
 
 static struct platform_driver brcmf_pd = {
 #if LINUX_VERSION_IS_GEQ(6,13,0)
 	.remove		= brcmf_common_pd_remove,
 #else
-	.remove = bp_brcmf_common_pd_remove,
+	.remove_new = brcmf_common_pd_remove,
 #endif
 	
 	.driver		= {
